src/machine.rs

```diff
@@ -695,8 +695,24 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for Evaluator<'mir, 'tcx> {
         ptr: Pointer<Self::PointerTag>,
     ) -> InterpResult<'tcx> {
         match ptr.provenance {
-            Tag::Concrete(concrete) =>
-                intptrcast::GlobalStateInner::expose_addr(ecx, concrete.alloc_id),
+            Tag::Concrete(ConcreteTag { alloc_id, sb }) => {
+                intptrcast::GlobalStateInner::expose_addr(ecx, alloc_id);
+
+                let (size, _) =
+                    ecx.get_alloc_size_and_align(alloc_id, AllocCheck::MaybeDead).unwrap();
+
+                // Function pointers and dead objects don't have an alloc_extra so we ignore them.
+                if let Ok(alloc_extra) = ecx.get_alloc_extra(alloc_id) {
```

Comment:
This "catches" an `InterpError`. Please don't do that; use `?` instead.

Reply:
I just tested this again, and the comment is still accurate. I need to filter out function pointers somehow, or else I get
```
Undefined Behavior: accessing alloc127 which contains a function
```
with the code being something like
```
action.sa_sigaction = signal_handler as sighandler_t
```

Reply:
Ah, right. The pointer might also dangle and this allocation no longer exist.

Catching `InterpError` is not the right solution, though. I don't have a much better idea right now... maybe putting the 'exposed' map into `AllocExtra` is not the right call after all. We can put it into `GlobalStateInner` for now and optimize this later if need be.